### PR TITLE
Implement ProtobufConvert for fixed byte arrays

### PR DESCRIPTION
**Pull request by alekseysidorov**
## Overview

---

See: https://jira.bf.local/browse/ECR-XYZW

### Definition of Done

- [x] There are no TODOs left in the merged code
- [x] Change is covered by automated tests
- [ ] Benchmark results are attached (if applicable)
- [x] The [coding guidelines] are followed
- [x] Public API has proper documentation
- [x] Changelog is updated if needed (in case of notable or breaking changes)
- [ ] The continuous integration build passes

[coding guidelines]: https://github.com/exonum/exonum/blob/master/CONTRIBUTING.md#conventions

### CODE REVIEW

File: exonum/src/proto/mod.rs [hunk 1]
```diff
@@ -338,3 +338,33 @@ impl_protobuf_convert_scalar!(i32);
...
+            /// Special case for fixed sized arrays.
+            impl ProtobufConvert for [u8; $arr_len] {
+                type ProtoStruct = Vec<u8>;
+
+                fn to_pb(&self) -> Self::ProtoStruct {
+                    self.to_vec()
+                }
+
+                fn from_pb(pb: Self::ProtoStruct) -> Result<Self, Error> {
+                    ensure!(pb.len() == $arr_len, "wrong array size: actual {}, expected {}", pb.len(), $arr_len);
```
Comment (dvush) [inline exonum/src/proto/mod.rs]: I think this line is too long.

File: exonum/src/proto/schema/exonum/tests.proto [hunk 2]
```diff
@@ -96,3 +96,9 @@ message TestProtobufConvertMap {
   map<uint32, Point> point_map = 5;
   map<string, uint64> key_string_map = 6;
 }
+
+message TestFixedArrays {
+  bytes fixed_array_8 = 1;
+  bytes fixed_array_16 = 2;
+  bytes fixed_array_32 = 3;
+}
```
Comment (dvush) [inline exonum/src/proto/schema/exonum/tests.proto]: '/n/ ?

File: exonum/src/proto/mod.rs [hunk 3]
```diff
@@ -338,3 +338,33 @@ impl_protobuf_convert_scalar!(i32);
...
+                        array
+                    })
+                }
+            }
+        )*
+    };
+}
+
+impl_protobuf_convert_fixed_byte_array! {
+    8, 16, 24, 32, 40, 48, 56, 64,
```
Comment (dvush) [inline exonum/src/proto/mod.rs]: Just curious, why this numbers?

Comment (aleksuss) [reply exonum/src/proto/mod.rs]: Most common lengths of arrays ? 🙃

**Pull request merged**